### PR TITLE
Allow modifying OCS CSV pre OCS deployment

### PR DESCRIPTION
**Pull request by ebenahar**
Allows usage of custom images for example, like MCG or RHCS

Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

**Comment by ebenahar** [comment]
```17:04:52 - MainThread - ocs_ci.ocs.ocp - INFO - Resource ocs-operator.v4.5.0-487.ci is in phase: Succeeded!
17:04:52 - MainThread - ocs_ci.utility.utils - INFO - CSV ocs-operator.v4.5.0-487.ci will be modified: quay.io/rhceph-dev/mcg-core@.* will be replaced with jeniawhite/noobaa-core:5.5-default-pool-custom-build.
The command that will be used for that is:
oc -n openshift-storage get csv -o yaml | sed 's,quay.io/rhceph-dev/mcg-core@.*,jeniawhite/noobaa-core:5.5-default-pool-custom-build,g' | oc replace -f -
17:04:52 - MainThread - ocs_ci.utility.utils - INFO - Executing command: chmod 777 /tmp/csv_modificationc2h0_89n.sh
17:04:52 - MainThread - ocs_ci.utility.utils - INFO - Executing command: sh /tmp/csv_modificationc2h0_89n.sh```

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 1]
```diff
@@ -158,6 +158,14 @@ def pytest_addoption(parser):
         dest='flexy_env_file',
         help="Path to flexy environment file"
     )
+    parser.addoption(
```
Comment (clacroix12) [inline ocs_ci/framework/pytest_customization/ocscilib.py:161]: Is this an option that should be repeatable or do we think that we will only want to change one part of the CSV?

Comment (ebenahar) [reply ocs_ci/framework/pytest_customization/ocscilib.py:161]: @clacroix12 I thought about making it possible to accept multiple values to replace but couldn't find a solution and also thought it doesn't worth investing time as up until now, we needed to change only 1 component at a time

Comment (clacroix12) [reply ocs_ci/framework/pytest_customization/ocscilib.py:161]: We definitely don't need to enhance this in this PR if it isn't necessary. That being said I don't think it will be difficult. If we make this parameter repeatable, the value of `--csv-change` would just end up with a list of values like this:

```python
["replace_me::replaced", "replace_me_too::also_replaced", ...]
```

The values being propagated to our config could then just be placed in a list format, and eventually iterated over when we go to replace the contents of the CSV with our replacement values.

I may be over-simplifying here or missing some part of the process that may cause complications but it seems like we could take this up in an enhancement if we saw the need to change multiple parts of the CSV for one execution.

Comment (ebenahar) [reply ocs_ci/framework/pytest_customization/ocscilib.py:161]: Discussed this offline and we will implement this in the future if it will be needed

File: conf/ocsci/io_in_bg.yaml [hunk 2]
```diff
@@ -1,4 +0,0 @@
----
-# This config file will enable IO to be running in the background
-RUN:
-  io_in_bg: True
```
Comment (vasukulkarni) [inline conf/ocsci/io_in_bg.yaml:4]: Is this unrelated cleanup?

Comment (ebenahar) [reply conf/ocsci/io_in_bg.yaml:4]: yes, it's unrelated. We don't need this file anymore

Comment (vasukulkarni) [reply conf/ocsci/io_in_bg.yaml:4]: Should have been in a different PR/commit.

File: ocs_ci/utility/utils.py [hunk 3]
```diff
@@ -2363,3 +2363,33 @@ def set_aws_region(region=None):
...
+
+    temp_file = NamedTemporaryFile(
+        mode='w+', prefix='csv_modification', suffix='.sh'
+    )
+
+    with open(temp_file.name, 'w') as t_file:
+        t_file.writelines(data)
+
+    run_cmd(f"chmod 777 {temp_file.name}")
+    run_cmd(f"sh {temp_file.name}")
```
Comment (vasukulkarni) [inline ocs_ci/utility/utils.py:2395]: why not run_oc_cmd ?

Comment (ebenahar) [reply ocs_ci/utility/utils.py:2395]: for some reason, while running this command with `run_cmd`, it doesn't work, I tried every possible way but only this way it works. 
Probably it has to do with `sed`

**Comment by vasukulkarni** [comment]
LGTM, Please squash into 2 commits (one doc fix and other csv change).

**Comment by ebenahar** [comment]
> LGTM, Please squash into 2 commits (one doc fix and other csv change).

Hi @vasukulkarni you mean to have 2 PRs?

**Comment by vasukulkarni** [comment]
> > LGTM, Please squash into 2 commits (one doc fix and other csv change).
> 
> Hi @vasukulkarni you mean to have 2 PRs?

It can be in this PR itself since its reviewed, just separating the commits using `git rebase -i`

**Comment by vasukulkarni** [comment]
There is a conflict and needs rebase.

File: ocs_ci/ocs/cluster_load.py [hunk 4]
```diff
@@ -334,33 +340,23 @@ def calc_trim_metric_mean(self, metric, samples=5, mute_logs=False):
             time.sleep(5)
         return round(get_trim_mean(vals), 5)
 
-    def get_metrics(self, mute_logs=False):
+    def print_metrics(self, mute_logs=False):
         """
-        Get different cluster load and utilization metrics
+        Print metrics
```
Comment (vasukulkarni) [inline ocs_ci/ocs/cluster_load.py:345]: typo?

Comment (vasukulkarni) [reply ocs_ci/ocs/cluster_load.py:345]: ignore,  the diff is confusing

Comment (ebenahar) [reply ocs_ci/ocs/cluster_load.py:345]: where?
